services-manager/src/main/java/uk/gov/gchq/palisade/service/manager/service/ManagedService.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.manager.service;
+
+import feign.Response;
+import feign.RetryableException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.gchq.palisade.service.Service;
+import uk.gov.gchq.palisade.service.manager.web.ManagedClient;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class ManagedService implements Service {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedService.class);
+    private final ManagedClient managedClient;
+    private final Supplier<Collection<URI>> uriSupplier;
+
+    public ManagedService(final ManagedClient managedClient, final Supplier<Collection<URI>> uriSupplier) {
+        this.managedClient = managedClient;
+        this.uriSupplier = uriSupplier;
+    }
+
+    public boolean isHealthy() {
+        Collection<URI> clientUris = this.uriSupplier.get();
+        return clientUris.stream()
+                .map(clientUri -> {
+                    int status = 404;
+                    try {
+                        status = this.managedClient.getHealth(clientUri).status();
+                    } catch (RetryableException ex) {
+                        // Not up yet
+                    }
+                    LOGGER.debug("Client uri {} has status {}", clientUri, status);
+                    return status;
+                })
+                // Could be allMatch, but only one healthy service is needed to perform requests
```

Comment:
true, but it surely needs to be the correct service for the request you want to send. Given I don't expect you know what the first request will be and therefore which service it requires. I think this should actually be allMatch

Reply:
Will change to allMatch